diff --git a/node_modules/@anchor-protocol/anchor.js/dist/facade/earn/earn.js b/node_modules/@anchor-protocol/anchor.js/dist/facade/earn/earn.js
index aa1a88d..8d4c188 100644
--- a/node_modules/@anchor-protocol/anchor.js/dist/facade/earn/earn.js
+++ b/node_modules/@anchor-protocol/anchor.js/dist/facade/earn/earn.js
@@ -37,7 +37,7 @@ class Earn {
                 address: getTotalDepositOption.address,
                 token_address: this._addressProvider.aTerra(getTotalDepositOption.market),
             })(this._addressProvider);
-            return new terra_js_1.Int(new terra_js_1.Dec(epochState.exchange_rate).mul(userATerraBalance.balance))
+            return new terra_js_1.Dec(new terra_js_1.Dec(epochState.exchange_rate).mul(userATerraBalance.balance))
                 .div(1000000)
                 .toString();
         });
diff --git a/node_modules/@anchor-protocol/anchor.js/dist/queries/terraswap/reverse-token-simulation.js b/node_modules/@anchor-protocol/anchor.js/dist/queries/terraswap/reverse-token-simulation.js
index ea56a88..55a7e43 100644
--- a/node_modules/@anchor-protocol/anchor.js/dist/queries/terraswap/reverse-token-simulation.js
+++ b/node_modules/@anchor-protocol/anchor.js/dist/queries/terraswap/reverse-token-simulation.js
@@ -16,7 +16,7 @@ const queryTerraswapReverseTokenSimulation = ({ lcd, contractAddr, amount, pair_
         reverse_simulation: {
             ask_asset: {
                 info: {
-                    ANC: {
+                    token: {
                         contract_addr: contractAddr,
                     },
                 },
diff --git a/node_modules/@anchor-protocol/anchor.js/dist/queries/terraswap/simulation.js b/node_modules/@anchor-protocol/anchor.js/dist/queries/terraswap/simulation.js
index f005d30..d779365 100644
--- a/node_modules/@anchor-protocol/anchor.js/dist/queries/terraswap/simulation.js
+++ b/node_modules/@anchor-protocol/anchor.js/dist/queries/terraswap/simulation.js
@@ -16,7 +16,7 @@ const queryTerraswapSimulation = ({ lcd, contractAddr, amount, pair_contract_add
         simulation: {
             offer_asset: {
                 info: {
-                    ANC: {
+                    token: {
                         contract_addr: contractAddr,
                     },
                 },
