diff --git a/node_modules/@terra-money/terra.js/dist/client/lcd/Wallet.d.ts b/node_modules/@terra-money/terra.js/dist/client/lcd/Wallet.d.ts
index a60432b..ebb7060 100644
--- a/node_modules/@terra-money/terra.js/dist/client/lcd/Wallet.d.ts
+++ b/node_modules/@terra-money/terra.js/dist/client/lcd/Wallet.d.ts
@@ -13,6 +13,6 @@ export declare class Wallet {
     }>;
     accountNumber(): Promise<number>;
     sequence(): Promise<number>;
-    createTx(options: CreateTxOptions): Promise<StdSignMsg>;
+    createTx(options: CreateTxOptions, address?: string): Promise<StdSignMsg>;
     createAndSignTx(options: CreateTxOptions): Promise<StdTx>;
 }
diff --git a/node_modules/@terra-money/terra.js/dist/client/lcd/Wallet.js b/node_modules/@terra-money/terra.js/dist/client/lcd/Wallet.js
index 201a583..e9fd444 100644
--- a/node_modules/@terra-money/terra.js/dist/client/lcd/Wallet.js
+++ b/node_modules/@terra-money/terra.js/dist/client/lcd/Wallet.js
@@ -60,8 +60,8 @@ var Wallet = /** @class */ (function () {
             return d.sequence;
         });
     };
-    Wallet.prototype.createTx = function (options) {
-        return this.lcd.tx.create(this.key.accAddress, options);
+    Wallet.prototype.createTx = function (options, address) {
+        return this.lcd.tx.create(address || this.key.accAddress, options);
     };
     Wallet.prototype.createAndSignTx = function (options) {
         return __awaiter(this, void 0, void 0, function () {
